tool/net/redbean.c

```diff
@@ -4638,6 +4638,31 @@ static int LuaSha512(lua_State *L) {
   return LuaHasher(L, 64, mbedtls_sha512_ret_512);
 }
 
+static noinline int LuaGetCryptoHash(lua_State *L) {
+  size_t hl, pl, kl;
+  uint8_t d[64];
+  mbedtls_md_context_t ctx;
+  // get hash name, payload, and key
+  void *h = luaL_checklstring(L, 1, &hl);
+  void *p = luaL_checklstring(L, 2, &pl);
+  void *k = luaL_optlstring(L, 3, "", &kl);
+
+  const mbedtls_md_info_t *digest = mbedtls_md_info_from_string(h);
```

Comment:
It's interesting how this worked even with the constness in the type signature. Strictly speaking you'd normally want to call md_init() and then md_setup() where you index into the result of md_info_from_string(). But looking at the code the way you did things should be fine, and would have been caught anyway by the `.rodata` protection had it been an issue.